librz/core/golang.c

```diff
@@ -1605,14 +1612,78 @@ static ut32 golang_recover_string_riscv64(GoStrRecover *ctx) {
 	return 4;
 }
 
+// Sometimes the data-structures has strings, but these are stored in tables where
+// the first offset is always the pointer to the squashed strings and the next word
+// is the size of the string
+static void core_recover_golang_strings_from_data_pointers(RzCore *core, GoStrRecover *ctx) {
+	rz_core_notify_begin(core, "Recovering go strings from the data sections");
+
+	RzAnalysis *analysis = core->analysis;
+	const ut32 word_size = analysis->bits / 8;
+	RzListIter *iter;
+	RzBinSection *section;
+	ut8 *buffer = NULL;
+	ut64 string_addr, string_size;
+	RzList *section_list = rz_bin_get_sections(core->bin);
+	if (!section_list) {
+		RZ_LOG_ERROR("Failed to get the RzBinSection list\n");
+		goto end;
+	}
+
+	buffer = malloc(GO_MAX_TABLE_SIZE);
+	if (!buffer) {
+		RZ_LOG_ERROR("Failed to allocate table buffer\n");
+		goto end;
+	}
+
+	rz_list_foreach (section_list, iter, section) {
+		if (section->vsize < (word_size * 2) || !strstr(section->name, "data")) {
+			continue;
+		}
+
+		ut64 current = section->vaddr;
+		ut64 end = section->vaddr + section->vsize;
+
+		do {
+			size_t length = RZ_MIN((end - current), GO_MAX_TABLE_SIZE);
+			if (length < (word_size * 2)) {
+				break;
+			}
+
+			if (0 > rz_io_nread_at(core->io, current, buffer, length)) {
+				RZ_LOG_ERROR("Failed to read section at address %" PFMT64x "\n", current);
+				break;
+			}
+
+			length -= word_size;
+			for (size_t i = 0; i < length; i += word_size) {
+				string_addr = rz_read_ble(buffer + i, analysis->big_endian, analysis->bits);
+				string_size = rz_read_ble(buffer + i + word_size, analysis->big_endian, analysis->bits);
+				if (!string_addr || !string_size) {
+					continue;
+				} else if (word_size == sizeof(ut32) && string_addr == UT32_MAX) {
```

Comment:
can you have the case word_size == sizeof(ut64)?

Reply:
yes, but is already handled by recover_string_at